lib/fbe/faraday_middleware/quota.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+# MIT License
+#
+# Copyright (c) 2024 Zerocracy
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+require 'faraday'
+
+# Faraday Middleware that monitors GitHub API rate limits.
+class Fbe::FaradayMiddleware::Quota < Faraday::Middleware
+  def initialize(app, options = {})
+    super(app)
+    @limit = 100
+    @requests = 0
+    @app = app
+    @logger = options[:logger]
```

Comment:
@Suban05 as far as I understand Ruby, it's better to specify these parameters in the declaration of `initialize` explicitly:

```
def initialize(app, logger: nil, github_api_pause: nil)
```

This way, the user will not be able to make a mistake with the name. Also, it will be obvious that `nil` is the default (which is a bad idea).